integration/tools/validation/src/main/java/alluxio/cli/EnvValidationTool.java

```diff
@@ -0,0 +1,376 @@
+      results.put(result.mState, results.getOrDefault(result, 0) + 1);
+      switch (result.mState) {
+        case OK:
+          System.out.print(Constants.ANSI_GREEN);
+          break;
+        case WARNING:
+          System.out.print(Constants.ANSI_YELLOW);
+          break;
+        case FAILED:
+          System.out.print(Constants.ANSI_RED);
+          break;
+        case SKIPPED:
+          System.out.print(Constants.ANSI_PURPLE);
+          break;
+        default:
+          break;
+      }
+      System.out.print(result.getName());
+      System.out.println(Constants.ANSI_RESET);
+      validationCount++;
+    }
+    if (results.containsKey(ValidationUtils.State.FAILED)) {
+      System.err.format("%d failures ", results.get(ValidationUtils.State.FAILED));
+    }
+    if (results.containsKey(ValidationUtils.State.WARNING)) {
+      System.err.format("%d warnings ", results.get(ValidationUtils.State.WARNING));
+    }
+    if (results.containsKey(ValidationUtils.State.SKIPPED)) {
+      System.err.format("%d skipped ", results.get(ValidationUtils.State.SKIPPED));
+    }
+    System.err.println();
+    if (validationCount == 0) {
+      System.err.format("No validation task matched name \"%s\".%n", name);
+      return false;
+    }
+    if (results.containsKey(ValidationUtils.State.FAILED)) {
+      return false;
+    }
+    System.out.println("Validation succeeded.");
+    return true;
+  }
+
+  /**
+   * Runs the worker validation tasks.
+   *
+   * @param name task name prefix
+   * @param cmd the command line
+   * @return whether the validations passed
+   * */
+  public boolean validateWorkers(String name, CommandLine cmd) throws InterruptedException {
+    return validateRemote(ConfigurationUtils.getWorkerHostnames(mConf), "worker", name, cmd);
+  }
+
+  /**
+   * Runs the master validation tasks.
+   *
+   * @param name target task prefix
+   * @param cmd the command line
+   * @return whether the validations passed
+   * */
+  public boolean validateMasters(String name, CommandLine cmd) throws InterruptedException {
+    return validateRemote(ConfigurationUtils.getMasterHostnames(mConf), "master", name, cmd);
+  }
+
+  private void printTasks(String target) {
+    System.out.format("The following tasks are available to run on %s:%n", target);
+    Collection<ValidationTask> tasks = mTargetTasks.get(target);
+    for (ValidationTask task: tasks) {
+      String taskName = mTasks.get(task);
+      System.out.printf("%s: %s%n", taskName, mTaskDescriptions.get(taskName));
+    }
+    System.out.println();
+  }
+
+  /**
+   * Prints all the tasks.
+   * */
+  public void printTasks() {
+    printTasks("master");
+    printTasks("worker");
+    printTasks("cluster");
+  }
+
+  /**
+   * Aggregates the options from each validation tasks.
+   * Each {@link ValidationTask} defines the {@link Options} in a OPTIONS field
+   *
+   * @return all options
+   * */
+  public static Options getOptions() {
```

Comment:
This replaces the old `ValidationTask#getOptions` method. This is because the options need to be known in the command line class, which is now separated into `shell/ValidateEnv` class. The options need to be known before initializing this `EnvValidationTool` class, which means we cannot register the options when we register tasks, we have to know the options before initializing the task objects. I chose to do a reflection loading the static `OPTIONS` field of each `ValidationTask` class here.

Subsequently the `ValidationTask#getOptions` method is removed, together with the `AbstractValidationTask` class, which only implements this method.

Reply:
not sure i understand the issue here, did the HmsTool have a similar problem?